### ISSUE TITLE
input images

### ISSUE THREAD

**Issue by hiroyamaeda** [thtrieu/darkflow#38]
Thank you for your great repository!
My question is not directly related to your repository, but what is darkflow & darknet input images?
Ex. input images should be normalized(0-255 -> 0-1)? 

Now, I'm trying to use yolo model made by this repository on Android(https://github.com/tensorflow/tensorflow/tree/master/tensorflow/examples/android/), but it seems to be something wrong...
My android output confidence is always nearly 1.

Sorry to bother you.

**Comment by thtrieu** [comment]
No problem :) Raw image (0-255) is the input **to the program**. However, for android usage, you would need to normalize image to (-1, 1) be the input of **the neural net**.

**Comment by hiroyamaeda** [comment]
Really thank you!!
This is my procedure(↓).
___
Procedure
1. Darknet & YOLO Model
- Model make and Learning
2. file convert.
- Using dartflow
- YOLO Model(.weights) -> TensorFlow (Protocol Buffer file)
3. Object detection
- TensorFlow on Android using YOLO Model. 
____

So please check my understanding,
1) if we want to use darknet model on tensorflow, we have to train darknet model by using normalized images(0-1).
2) (-1, 1) is a mistake? I think it is (0, 1)..

thank you for your corperation!!

**Comment by thtrieu** [comment]
I double-checked, you are right, (0, 1) is the normalize range.

To run on Android with trained model, you don't need to train anything.

1. Download weight files from YOLO homepage
2. Put weight file to `bin/your_weight_file.weights`
4. Run `./flow --model cfg/corresponding_cfg_file.cfg --load bin/your_weight_file.weights --savepb`
5. Then you will obtain a `.pb` file, put this into your android application
6. On your android application, you have to pre-process and post-process yourself, I can see in the Android demo of Tensorflow [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/android/src/org/tensorflow/demo/TensorFlowYoloDetector.java), they've done it already.

**Comment by thtrieu** [comment]
I double-checked, you are right, (0, 1) is the normalize range.

To run on Android with trained model, you don't need to train anything.

1. Download weight files from YOLO homepage
2. Put weight file to `bin/your_weight_file.weights`
4. Run `./flow --model cfg/corresponding_cfg_file.cfg --load bin/your_weight_file.weights --savepb`
5. Then you will obtain a `.pb` file, put this into your android application
6. On your android application, you have to pre-process and post-process yourself, I can see in the Android demo of Tensorflow [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/android/src/org/tensorflow/demo/TensorFlowYoloDetector.java), they've done it already.

**Comment by hiroyamaeda** [comment]
Finally, I succeeded!
Thank you for your help!☺️

**Comment by thtrieu** [comment]
Congratulations, do not hesitate to ask for help here.

**Comment by abhigoku10** [comment]
@thtrieu @hiroyamaeda  i performed the same steps indicated by you but after converting into .pb i put the file into the android yolo repository and using it i get a common error " Could not create TensorFlow Graph: Not found: " do u know what might be the issue

**Comment by shenyingying** [comment]
> Finally, I succeeded!
> Thank you for your help!
@hiroyamaeda  you use yolov2 or yolov2-tiny ?
